tc/core/polyhedral/body.h

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2018, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <iostream>
+
+#include "tc/external/isl.h"
+
+namespace tc {
+namespace polyhedral {
+
+// Information about the bodies of the polyhedral statements.
+struct Body {
+  Body() = default;
```

Comment:
I'm not entirely sure we want this class to be default-constructible, but it is apparently necessary in the code

Reply:
It's prob because `result.body = body;` above.
We should be able to drop the default constructor and just:
 `return ScheduleTreeAndAccesses{body, fromIslSchedule(schedule)};`

Reply:
On Fri, Jun 29, 2018 at 08:52:39AM -0700, Nicolas Vasilache wrote:
> It's prob because `result.body = body;` above.

No, it's the default constructor.  It's needed   
because both `ScheduleTreeAndAccesses` and `Scop`
have default constructors.

> We should be able to drop the default constructor and just:
>  `return ScheduleTreeAndAccesses{body, fromIslSchedule(schedule)};`

`ScheduleTreeAndAccesses` has more than two fields,
so this wouldn't work.